### PR TITLE
v4 NFT docs

### LINKED ISSUE #701: Add deployment version, copy OEP markdowns
**Is your feature request related to a problem? Please describe.**
To do:
- [x] Move READMEs from https://github.com/oceanprotocol/oeps to docs website
- [x] Create Documentation related to upcoming v4 + NFT (explain NFTs in general and show Ocean Protocol will support those)
- [x] Create a Deployments page to show in which network what version of contracts, Aquarius, Provider, etc is running.

### LINKED ISSUE #781: Create v4 NFT docs
**Is your feature request related to a problem? Please describe.**
Create docs for the upcoming v4 release.
- [x] Introduction page
- [x] Use case: Sub-licensing
- [x] Roles and permissions page
- [x] Architecture diagram

Update:

- Removed: Use case: Transfer Base IP
- Change check box char for Roles page, add some text context

**Describe the solution you'd like**
Beginner friendly docs explaining the v4 contracts, use cases, flows,etc.
These docs can later explain advanced use cases.

### PR DESCRIPTION
**Pull request by akshay-ap**
Fixes #781 

Changes proposed in this PR:

- Create an introductory page for NFT + v4
- Create a roles page

### CODE REVIEW

**Comment by akshay-ap** [comment]
https://deploy-preview-780--docs-oceanprotocol.netlify.app/concepts/v4-nft/

**Comment by akshay-ap** [comment]
https://deploy-preview-780--docs-oceanprotocol.netlify.app/concepts/v4-roles/

File: data/sidebars/concepts.yml [hunk 1]
```diff
@@ -25,6 +25,13 @@
     - title: DDO Metadata
       link: /concepts/ddo-metadata/
 
+- group: NFTs (upcoming v4 release)
```
Comment (kremalicious) [inline data/sidebars/concepts.yml:28]: Why is this made so important? We do not have a dedicated datatoken section, so why have a dedicated NFT section? It should all be part of exisitng sections, like “Specifiying assets”

File: content/concepts/v4-nft.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (kremalicious) [inline content/concepts/v4-nft.md:1]: please remove version references from all URLs. It makes it harder moving forward with a next version if we do that. We also dont have our old stuff under `v2` URLs

File: content/concepts/nft.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+A non-fungible token stored on the blockchain represents a unique asset. NFTs can represent images, videos, digital art, or any piece of information. NFTs can be traded and allow transfer of ownership and copyright. [EIP-721](https://eips.ethereum.org/EIPS/eip-721) defines an interface for handling NFTs on Ethereum blockchain. The creator of the NFT can deploy a new contract on Ethereum or any Blockchain supporting NFT related interface and also, transfer the ownership through transfer transactions.
+
+## Terminology
+
+- **Base IP** means the artifact being copyrighted. Represented by the {ERC721 address, tokenId} from the publish transactions.
+- **Base IP holder** means the holder of the Base IP. Represented as the the actor that did the initial "publish" action.
+- **Sub-licensee** is the holder of the sub-license. Represented as the entity that controls address ERC721.\_owners[tokenId=x].
+- **To Publish**: Claim copyright or exclusive base license
+- **To Sub-license**:Transfer one (of many) sub-licenses to new licensee: ERC20.transfer(to=licensee, value=1.0)
```
Comment (kremalicious) [inline content/concepts/nft.md:14]: all this talk in here about "sub-licenses" but we never mention datatokens? I changed this in the overall page description at top and I strongly suggest if sub-licensing or managing access to a dataset with data tokens is meant with "sub-licensing", then it should be made way more clear throughout whole document

File: content/concepts/nft.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+title: Data NFTs
+description: Ocean Protocol uses ERC721 NFTs to represent the intellectual property (IP) of published datasets or algorithms, and to further sub-license access to it with user roles or ERC20 data tokens.
+---
+
+A non-fungible token stored on the blockchain represents a unique asset. NFTs can represent images, videos, digital art, or any piece of information. NFTs can be traded and allow transfer of ownership and copyright. [EIP-721](https://eips.ethereum.org/EIPS/eip-721) defines an interface for handling NFTs on Ethereum blockchain. The creator of the NFT can deploy a new contract on Ethereum or any Blockchain supporting NFT related interface and also, transfer the ownership through transfer transactions.
+
+## Terminology
+
+- **Base IP** means the artifact being copyrighted. Represented by the {ERC721 address, tokenId} from the publish transactions.
```
Comment (kremalicious) [inline content/concepts/nft.md:10]: asset, dataset, algorithm, artifact. We use a lot of different names for the same thing so in this case let's just stick to "asset", and NOT introduce yet another term with "artifact".

Also "Base IP" feels like a confusing term if we never write out "IP". Is it "Intellectual Property"?

**Comment by kremalicious** [comment]
Some quotes from various meetings describing the relationship between NFT & datokens we might want to use in here.

@trentmc correct me if I'm wrong with the following blurbs but trying to get the very dry language of this doc draft introducing new terms into something easily readable and understandable. We now have the challenge that people got used to having the datatoken, and its symbol, as the (unique, not technically but de facto) representation for each asset. [Rugpullindex](https://rugpullindex.com) & [Alga app](https://twitter.com/realdatawhale/status/1458140133027246082) as case in point. So we need to make the relationship between a newly introduced token now actually uniquely representing the dataset really clear as quickly [outlined here](https://github.com/oceanprotocol/docs/pull/780#discussion_r745458880).

> Data NFTs represent the intellectual property over an asset, while datatokens attached to asset services define and allow access to the asset.

> While the NFT could be traded outside of Ocean Market and therefore itself being priced, the actual asset price for download/compute continues to be defined by an attached datatoken/OCEAN pool for pool assets.

> The NFT represents the ownership over an asset, whoever has the NFT of an asset owns it in the blockchain sense and in most Ocean dApps will be detected as the publisher. Lawyers won't like this statement. The owner is able to define various roles within the NFT, allowing to delegate actions like updating metadata to managers.

> Datatokens define the access to an asset, whoever has a datatoken can download or start a compute job on an asset. They are bound to download or compute services defined and stored within the DDO. As there can be multiple services for an asset, there can be multiple datatokens for an asset, and ultimately multiple pools for an asset. The Ocean Market UI allows only one service and one pool per asset in its first release, but forks are free to extend on that.

**Comment by kremalicious** [comment]
- mention datatoken name & symbol generation (will stay as is)
- mention NFT name & symbol generation (needs a new concept first)

File: content/concepts/architecture.md [hunk 5]
```diff
@@ -11,21 +11,31 @@ Here is the Ocean architecture.
 
 Here’s an overview of the figure.
 
-- The top layer is **applications** like Ocean Market. With these apps, users can onboard data services into crypto (publish and mint datatokens), hold datatokens as assets (data wallets), discover data assets and buy / sell datatokens for fixed or auto-determined price (data marketplaces), and consume data services (consume datatokens).
-- Below that are **libraries** used by the applications: Ocean React hooks, JavaScript library, and Python library. This also includes middleware to assist discovery: Aquarius and (3rd party tool) TheGraph.
-- The lowest level has the **smart contracts** used by the libraries. They’re deployed on Ethereum mainnet to start, and other networks later.
+- The top layer is **applications** like Ocean Market. With these apps, users can onboard services like data, alogrithm, compute-to-data into crypto (publish and mint DataNFTs and Datatokens), hold datatokens as assets (data wallets), discover assets and buy / sell datatokens for fixed or auto-determined price (data marketplaces), and consume data services (consume datatokens).
```
Comment (trentmc) [inline content/concepts/architecture.md:14]: - "DataNFTs" -> "dataNFTs"
- "Datatokens" -> "datatokens"

In general, only capitalize if at the beginning of a sentence or in a heading.

File: content/concepts/architecture.md [hunk 6]
```diff
@@ -11,21 +11,31 @@ Here is the Ocean architecture.
...
+- Below that are **libraries** used by the applications: Ocean.js (JavaScript library) and Ocean.py (Python library). This also includes middleware to assist discovery:
+    - **Aquarius**: Provides metadata cache for faster serach by caching on-chain data into elasticsearch
```
Comment (trentmc) [inline content/concepts/architecture.md:16]: "serach" -> "search"

File: content/concepts/architecture.md [hunk 7]
```diff
@@ -59,9 +69,9 @@ Complementary to Ocean Market, Ocean has reference code to ease building **third
...
 
-Ocean uses the Ethereum mainnet as an **on-chain metadata store**, i.e. to store both DID and DDO. This means that once the write fee is paid, there are no further expenses or dev-ops work needed to ensure metadata availability into the future, aiding in the discoverability of data assets. It also simplifies integration with the rest of the Ocean system, which is Ethereum-based. Storage cost on Ethereum mainnet is not negligible, but not prohibitive and the other benefits are currently worth the tradeoff compared to alternatives.
+Ocean uses the Ethereum mainnet and other compatible networks as an **on-chain metadata store**, i.e. to store both DID and DDO. This means that once the write fee is paid, there are no further expenses or dev-ops work needed to ensure metadata availability into the future, aiding in the discoverability of data assets. It also simplifies integration with the rest of the Ocean system, which is Ethereum-based. Storage cost on Ethereum mainnet is not negligible, but not prohibitive and the other benefits are currently worth the tradeoff compared to alternatives.
```
Comment (trentmc) [inline content/concepts/architecture.md:74]: Replace paragraph with:

"""Metadata is stored by the Publisher, alongside data. This allows for personally identifiable information (PII) to be removed (unlike if metadata was on-chain)."""

File: content/concepts/data-nft.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+### Manager
+
+- A public address with a `Magner` role can update the metadata
+- Can deploy new ERC20 contract which is associtated with the `ERC721` contract.
+
+## Use case 1
+
+Alice is the creator of a painting and wants to make sure that the ownership of her artwork is uniquely determined. She performs the **Publish** action. She creates an ERC721 token which represents the ownership of the physical asset. Here, **Base IP** is Alice's artwork. **Base IP holder** is Alice. Now, Alice wants to transfer the ownership of her Artwork to Bob. So, Alice makes Bob's address as owner of the token in the discussion by sending a Blockchain transaction.
+
+![Image 1](images/use-case-1.PNG)
```
Comment (trentmc) [inline content/concepts/data-nft.md:45]: Delete all of "Use case 1". It's not how Ocean uses ERC721.

File: content/concepts/data-nft.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+## Use case 2
+
+Alice is the author of a book. Alice wants to hold the copyright of her work but, allows others to read her book. So, She creates 2 versions of her book namely: digital copy, physical copy. She assigns Bob as one of the holder of digital edition of the book. Here, **Base IP** is the book. **Base IP holder** is Alice and Bob is **Sub-licensee** for a digital edition of the book. Alice tokenizes her work by performing **Publish** action i.e _ERC721.safeMint(to=aliceWalletAddress, tokenid=1)_. Alice also creates sub-licenses of her book by creating two ERC20 tokens and transfers the digital copy token to Bob's wallet.
```
Comment (trentmc) [inline content/concepts/data-nft.md:49]: Replace with the following. (Keep the same image.)

"""
## High-Level Behavior

Here's an example. Alice has created a dataset. In step 1, she **publishes** her dataset with Ocean: this means deploying an ERC721 dataNFT contract (claiming copyright / base IP), then an ERC20 datatoken contract (license against base IP). Then in step 2, she mints some ERC20 datatokens and transfers 1.0 of them to Bob's wallet; now he has a license to be able to consume that dataset.
""""

File: content/concepts/data-nft.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+![Image 2](images/use-case-2.PNG)
+
+## Other References
+
+- [Practical Connections of ERC721 with Intellectual Property](https://blog.oceanprotocol.com/nfts-ip-1-practical-connections-of-erc721-with-intellectual-property-dc216aaf005d)
```
Comment (trentmc) [inline content/concepts/data-nft.md:55]: Add "Data & NFTs 1:" in front of this bullet.

Do the same for the other two: "Data & NFTs 2", and "3".

**Review by trentmc — COMMENTED** [review]
Hi @akshay-ap, sorry for the delay, I've now reviewed and suggested edits. Thanks!

File: content/concepts/architecture.md [hunk 11]
```diff
@@ -11,21 +11,31 @@ Here is the Ocean architecture.
...
+- Below that are **libraries** used by the applications: Ocean.js (JavaScript library) and Ocean.py (Python library). This also includes middleware to assist discovery:
+    - **Aquarius**: Provides metadata cache for faster serach by caching on-chain data into elasticsearch
+    - **Provider**: Facilitates downloading assets, DDO encryption and communicating with `operater-service` for Compute-to-Data jobs.
+    - **TheGraph**: 3rd party tool
+Developers can utilize the libraries to built thier custom applications and marketplaces.
+- The lowest level has the **smart contracts** used by the libraries. They’re deployed on Ethereum mainnet, and other compatible networks. To see the list of supported networks click [here](/concepts/networks/).
 
-Left to right are groupings of functionality: tools for datatokens, tools for markets (including pools), tools to consume data services and for metadata, and external ERC20 tools.
+Left to right are groupings of functionality: tools for datatokens, tools for markets (including pools), tools to consume data services and for metadata, and external ERC20, ERC721 tools.
```
[the remark this answers is not in the document]
Comment (kremalicious) [reply content/concepts/architecture.md:22]: Please don't use Datatokens & ERC20  when meaning the same thing. We do not have "ERC20 tools", but datatoken tools if you want to name them. Saying something like ERC20 tools implies we have tools to create any ERC20, when in fact we only want people to create datatokens for access. Everything else is misleading

File: content/concepts/architecture.md (hunk 7 above)
Comment (kremalicious) [reply content/concepts/architecture.md:74]: > Metadata is stored by the Publisher, alongside data. 

nope, metadata is stored on-chain in the NFT contract just as existing paragraph says.

File: content/concepts/architecture.md [hunk 12]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Architecture Overview
-description: Simplicity and Interoperability via a Datatokens Core
+description: Simplicity and Interoperability via a DataNFT Core
```
[the remark this answers is not in the document]
Comment (kremalicious) [reply content/concepts/architecture.md:3]: So "Simplicity and Interoperability via a core of DataNFTs and Datatokens" is not really describing what we do, most of the things we have are the opposite of simplicity. This sentence here should summarize what benefit our architecture provides to the reader.

Likewise, not sure about the formatting of "DataNFT". Putting a word and an abbreviation together and call it a word is just wrong. It's "Data NFT".

File: content/concepts/nft.md (hunk 3 above)
Comment (trentmc) [reply content/concepts/nft.md]: Agreed. Click on the "Files changed" tab and you'll see specific recommendations that I'm making on exactly this.

File: content/concepts/nft.md (hunk 4 above)
Comment (trentmc) [reply content/concepts/nft.md]: Agreed. Click on the "Files changed" tab and you'll see specific recommendations that I'm making on exactly this.

File: content/concepts/architecture.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (trentmc) [reply content/concepts/architecture.md:3]: > DataNFT [vs Data NFT]

I was on the fence on this while I reviewed this morning. You make good points. OK, I'm fine with "Data NFT". Would love to know if @alexcos20 has thoughts.

>  "Simplicity and Interoperability via a core of DataNFTs and Datatokens" 
> This sentence here should summarize what benefit our architecture provides to the reader.

How about "Data NFTs and Datatokens architecture"

File: content/concepts/architecture.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (trentmc) [reply content/concepts/architecture.md:22]: Good point. My first thought was to simply remove the last few words.

Even better, I see now, let's remove the whole paragraph. Why: while it was useful for Ocean v3, but Ocean v4 architecture doesn't group as cleanly, and the paragraph doesn't help.

File: content/concepts/architecture.md (hunk 7 above)
Comment (trentmc) [reply content/concepts/architecture.md:74]: @alexcos20 please resolve. Is metadata on-chain or off?

Comment (kremalicious) [reply content/concepts/architecture.md:74]: see https://github.com/oceanprotocol/docs/blob/feature/ddo_v4/content/concepts/did-ddo.md#rules-for-dids--ddos

**Review by trentmc — APPROVED** [review]

**Pull request merged**